code/addons/a11y/src/components/VisionSimulator.tsx

```diff
@@ -121,19 +112,39 @@ const getColorList = (active: Filter, set: (i: Filter) => void): Link[] => [
   }),
 ];
 
+const hiddenStyles = {
+  position: 'absolute',
+  width: '0',
+  height: '0',
+};
+
+const injectFilters = (iframeDocument: Document) => {
+  if (!iframeDocument.getElementById('color-filters')) {
+    const filtersContainer = iframeDocument.createElement('div');
+    filtersContainer.id = 'color-filters';
+    Object.assign(filtersContainer.style, hiddenStyles);
+    iframeDocument.body.appendChild(filtersContainer);
+    const filtersSvgString = ReactDOMServer.renderToStaticMarkup(<Filters />);
+    filtersContainer.innerHTML = filtersSvgString;
+  }
+};
+
 export const VisionSimulator = () => {
   const [filter, setFilter] = useState<Filter>(null);
+
+  useEffect(() => {
+    const iframe = document.getElementById(iframeId) as HTMLIFrameElement;
+    if (iframe && iframe.contentDocument) {
```

Comment:
Accessing the contentDocument will only work if the iframe is on the same origin, which is not always the case.

We should encapsulate this behavior in a storybook decorator instead, similar to how the backgrounds addon works.

Syncing state via a global.

https://github.com/storybookjs/storybook/blob/513bb66bb9729ece57581b6eb50e5b338c47c0b9/code/addons/backgrounds/src/decorator.ts#L23

Reply:
Hi @ndelangen thank you for taking a look at our PR and for leaving such a detailed response.

In the issue page, we left a simpler alternate solution which is to remove the seemingly unnecessary boxShadow css in iframe.tsx. The boxShadow has no visual effect on preview window at the moment and since the blur filter only causes the bug with this boxShadow, this would resolve the weird border bug and then upgrading the addon can be a separate issue? You can find this here:

https://github.com/storybookjs/storybook/blob/4bbfb143c4911eaff21293499ab914c92b0fae44/code/core/src/manager/components/preview/Iframe.tsx#L17

If you still prefer upgrading the addon for this particular issue, could you roughly outline the steps needed and your expectation? Like would the a11y addon need to be structured similarly to the background addon? We would like to understand this approach a bit better before diving right in.